lib/mix/lib/releases/assembler.ex

```diff
@@ -497,6 +505,12 @@ defmodule Mix.Releases.Assembler do
       do: :ok
   end
 
+  defp generate_cookie do
+    # 255 is max. atom length, see http://erlang.org/doc/efficiency_guide/advanced.html
+    # Cookies, at least when read from disk, need to be from the printable ASCII range
+    (1..255)|> Enum.map(fn _ -> Enum.random(?\s..?~) end) |> List.to_atom
```

Comment:
I don't know about OpenSSL availability on all target platforms, maybe somebody can chime in here and give insights.
Re (2), I'd user `strong_rand_bytes` and then transform to the printable range.

Regarding accessibility from the internet, one point that is often forgotten is a developer's machine. While less of an interesting target in itself, a developer might leave his/her local machine and node/cluster attackable for remote code execution.

So the approach should be for now: as high an entropy we can guarantee, while keeping portability.